libr/bin/p/bin_bflt.c

```diff
@@ -144,20 +144,19 @@ static int get_ngot_entries(struct r_bin_bflt_obj *obj) {
 static RList *relocs(RBinFile *bf) {
 	struct r_bin_bflt_obj *obj = (struct r_bin_bflt_obj *) bf->o->bin_obj;
 	RList *list = r_list_newf ((RListFree) free);
-	int i, len, n_got, amount;
+	ut32 i, len, n_got, amount;
 	if (!list || !obj) {
 		r_list_free (list);
 		return NULL;
 	}
 	if (obj->hdr->flags & FLAT_FLAG_GOTPIC) {
 		n_got = get_ngot_entries (obj);
 		if (n_got) {
-			amount = n_got * sizeof (ut32);
-			if (amount < n_got || amount > UT32_MAX) {
+			if (n_got > UT32_MAX / sizeof (struct reloc_struct_t)) {
 				goto out_error;
 			}
-			struct reloc_struct_t *got_table = calloc (
-				1, n_got * sizeof (struct reloc_struct_t));
+			amount = n_got * sizeof (struct reloc_struct_t);
+			struct reloc_struct_t *got_table = calloc (1, amount);
```

Comment:
same

Reply:
just noticed this after the 2 aprovals. can you have a look at it?

Reply:
`if(ngot)`
guaranteed ngot>=1 and amount >=1